File: Practice/Projects/team_game.py
# MY and AC, P6

import random

#Win check function, written by MY
def who_win(X, Y, Z):
    if X == "x" and Y == "x" and Z == "x": 
        #then player won
        return 1
    elif X == "o" and Y == "o" and Z == "o":
        #then computer won
        return 2
    else:
        return 0
    
#board print function, written by MY
def printboard():
    print(f"{s_list[0]}|{s_list[1]}|{s_list[2]}")
    print("----")
    print(f"{s_list[3]}|{s_list[4]}|{s_list[5]}")
    print("-----")
    print(f"{s_list[6]}|{s_list[7]}|{s_list[8]}")

#tie checker, written by MY
def check(checked):
    if type(s_list[checked]) == str:
        return True


#set up zone, written by MY
print("All inputs must be intigers")

s_list = [1, 2, 3, 4, 5, 6, 7, 8, 9]

winner = 0 

#central loop, written MY
while winner == 0:
    #Board print zone
    printboard()
    
#player answer zone, written by MY

    

    playerlooplock = 0

    while playerlooplock == 0:
        choice = int(input("Which place would you like to change?: "))-1

        if s_list[choice] == "o" or s_list[choice] == "x": 
            print("That spot is taken, try again.")
        else:
            playerlooplock += 1


    s_list[choice] = "x"

#computer answer zone, written by MY and AC

    computer_looplock = 0

    while computer_looplock == 0:
      #random.choice returns the item itself not the index # of the item
        computer_choice = random.choice(s_list)
    
        if not computer_choice == "o" and not computer_choice == "x": 
            computer_looplock += 1
            s_list[computer_choice-1] = "o"
        elif check(0) == True and check(1) == True and check(2) == True and check(3) == True and check(4) == True and check(5) == True and check(6) == True and check(7) == True and check(8) == True:
            computer_looplock += 1
            print("Tie!")
    
    
   

#Win check, written by AC                         
    winner += who_win( s_list[0], s_list[1], s_list[2])
    
    winner += who_win( s_list[3], s_list[4], s_list[5])

    winner += who_win( s_list[6], s_list[7], s_list[8])

    winner += who_win( s_list[0], s_list[4], s_list[8])

    winner += who_win( s_list[2], s_list[7], s_list[8])

    winner += who_win( s_list[0], s_list[3], s_list[6])

    winner += who_win( s_list[1], s_list[4], s_list[7])

    winner += who_win( s_list[2], s_list[5], s_list[8])    

#If this line is reached game loop has been broken becuse somone won.



#game win results, written by MY

if winner == 1:
    #then player won
    print("Congradulations you win!")

    printboard()

    
elif winner == 2:
    #then com won
    print("YOU LOSE!")

    printboard()

File: C/notes/projects/team_game.c
//P6, MY DS, Programming Project - C
#include <stdio.h>
#include <string.h>
#include <stdbool.h>
#include <math.h>

//For random
#include <time.h>
#include <stdlib.h>

//Maybe if I feel funny I will change the win condition to the player being in a tie -MY

//global varibal set up

char board_state[9] = {'0', '1', '2', '3', '4', '5', '6', '7', '8'};
bool anyone_won = false;

bool player_won = false;
bool computer_won = false;
bool tie = false;

int total_computer_tries = 0;

// Win check funciton made by MY

void winner_finder(char first_in_row, char second_in_row, char thrird_in_row){
    if ( first_in_row == 'X' && second_in_row == 'X' && thrird_in_row == 'X'){
        //Then player won
        player_won = true;
        anyone_won = true;
    }
    else if ( first_in_row == 'O' && second_in_row == 'O' && thrird_in_row == 'O'){
        //Then computer won
        computer_won = true;
        anyone_won = true;
    }
}

///Board print function written by MY

void printboard(){
    printf("%c|%c|%c\n", board_state[0], board_state[1], board_state[2]);
    printf("-----\n");
    printf("%c|%c|%c\n", board_state[3], board_state[4], board_state[5]);
    printf("-----\n");
    printf("%c|%c|%c\n",board_state[6], board_state[7], board_state[8]);
}



int main(){
    //Set up zone by MY
    

    printf("Welcome to Tic Tac Toe. There are some things you must know.\n");
    printf("All inputs must be intigers, else the machine spirit residing inside your computer will try to rebel.\n");
    printf("There may be more.\n");
    
    //Primary Loop made by MY

    

    while (anyone_won == false){

        printf("\n");
        printboard();

        //Player answer zone made by My
        bool valid_player_answer = false;

        int player_choice;
        while(valid_player_answer == false){
            

            printf("Which place would you like to change?: \n");
            scanf("%d", &player_choice);

            if(board_state[player_choice] == 'O' || board_state[player_choice] == 'X'|| player_choice < 0 || player_choice > 8){
                printf("INVALID, try again.\n");

            }
            else{
                board_state[player_choice] = 'X';
                valid_player_answer = true;
            }
        }

            

        //Computer answer zone written by MY
        bool valid_computer_choice = false;
        int computer_tries = 0;

        while(valid_computer_choice == false){
            int computer_choice = rand() %9; 
            if(computer_tries > 34){
                tie = true;
                anyone_won = true;
                valid_computer_choice = true;
            }
            else if (board_state[computer_choice] == 'X' || board_state[computer_choice] == 'O'){
                //Nothing Really happens
                computer_tries++;
            }
           
        
            else{
                valid_computer_choice = true;
                board_state[computer_choice] = 'O';
            }
        }
        if (computer_tries > 0){
                printf("\nFUN FACT IT TOOK THE COMPUTER %d ATTEMPT(S) TO FIND A VALID SPACE\n", computer_tries);
                total_computer_tries = total_computer_tries + computer_tries;
            }

        //Win Conditions by DS
        winner_finder(board_state[0],board_state[1], board_state[2]);
        winner_finder(board_state[3],board_state[4], board_state[5]);
        winner_finder(board_state[6],board_state[7], board_state[8]);
        winner_finder(board_state[0],board_state[4], board_state[8]);
        winner_finder(board_state[2],board_state[4], board_state[6]);
        winner_finder(board_state[0],board_state[3], board_state[6]);
        winner_finder(board_state[1],board_state[4], board_state[7]);
        winner_finder(board_state[2],board_state[5], board_state[8]);



        
    }
    printf("\nDEV NOTIFICATION: PRIMARY WHILE LOOP BROKE\n");
    //Win results zone made by DS
    if (player_won == true){
        printf("\n");
        printboard();
        printf("\nYou win");
    }

    else if (computer_won==true){
        printf("\n");
        printboard();
        printf("\nYou lost");
    }
    else if (tie == true){
        printf("\n");
        printboard();
        printf("\nTHE COMPUTER HAS SPOKEN\nIT GAVE UP\n You win by defult");
    }
    // FUN FACT made by MY
    printf("\n\nFUN FACT: IT TOOK THE COMPUTER A TOTAL OF %d ATTEMPTS TO FIND A VALID SPACE FOR THIS GAME TO END.", total_computer_tries);
    return 0;
}